### PR TITLE
Fix v9 Auth types 

### PR DESCRIPTION
**Pull request by Feiyang1**
- Remove internal types in the public typings
- Export public types which were not exported before
- Move const enums out of `public-types.ts` since they are not part of the public API anymore

Fixes https://github.com/firebase/firebase-js-sdk/issues/5256

### CODE REVIEW

File: packages-exp/auth-exp/src/model/enums.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Enumeration of supported providers.
+ *
+ */
+export const enum ProviderId {
```
Comment (Feiyang1) [inline packages-exp/auth-exp/src/model/enums.ts:22]: @sam-gc Let me if you'd like to put them in different places.

**Review by sam-gc — APPROVED** [review]

File: packages-exp/auth-exp/src/model/public_types.ts [hunk 2]
```diff
@@ -127,6 +64,22 @@ export interface Config {
   authDomain?: string;
 }
 
+/**
+ *
+ * reCAPTCHA parameters
```
Comment (Feiyang1) [inline packages-exp/auth-exp/src/model/public_types.ts:69]: @egilmorez this is a new interface we are exporting

Comment (Feiyang1) [reply packages-exp/auth-exp/src/model/public_types.ts:69]: @kevinthecheung Can you please take a look at it?

File: packages-exp/auth-exp/src/model/public_types.ts [hunk 3]
```diff
@@ -127,6 +64,22 @@ export interface Config {
...
+ *
+ * Check the reCAPTCHA docs for a comprehensive list. All parameters are accepted except for
+ * the sitekey. Firebase Auth backend provisions a reCAPTCHA for each project and will
+ * configure this upon rendering. For an invisible reCAPTCHA, a size key must have the value
+ * 'invisible'.
+ *
+ * @public
+ */
```
Comment (kevinthecheung) [inline packages-exp/auth-exp/src/model/public_types.ts:77]:
```suggestion
/**
 * Interface representing reCAPTCHA parameters.
 *
 * See the [reCAPTCHA docs](https://developers.google.com/recaptcha/docs/display#render_param) 
 * for the list of accepted parameters. All parameters are accepted except for `sitekey`: Firebase Auth 
 * provisions a reCAPTCHA for each project and will configure the site key upon rendering.
 *
 * For an invisible reCAPTCHA, set the `size` key to `invisible`.
 *
 * @public
 */
```

**Review by kevinthecheung — APPROVED** [review]

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**